enhancements/ingress/set-replace-append-delete-deny-headers.md

```diff
@@ -0,0 +1,491 @@
+   Adding a second way to configure HSTS would significantly complicate this enhancement.
+
+### Non-Goals
+
+1. Validation that each specified value is valid for the respective header (e.g. require that either "DENY" or "SAMEORIGIN" is 
+   specified for [the `x-frame-options` header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options)).
+2. Setting load balancing algorithm using header values.
+   This request is orthogonal to customizing HTTP headers. It is out of scope for this enhancement and should be handled
+   with a seprate RFE.
+3. Configuring buffer sizes by referring headers. Eg- tune.bufsize and tune.maxrewrite
+4. Modifying headers for a passthrough route.
+5. [Replace part of a header by using a regular expression](https://www.haproxy.com/documentation/hapee/latest/traffic-routing/rewrites/rewrite-responses/#replace-part-of-a-header-by-using-a-regular-expression) 
+6. [Replace a header by using a regular expression](https://www.haproxy.com/documentation/hapee/latest/traffic-routing/rewrites/rewrite-responses/#replace-a-header-by-using-a-regular-expression)
+
+## Proposal
+
+### Workflow Description
+
+**Cluster admin** is a human user responsible for deploying a
+cluster.
+
+**Users** is a human user responsible for developing and
+deploying an application in a cluster.
+1. Cluster admin wants to set/delete HTTP response headers for all the routes of an Ingress Controller.
+   Cluster admin edits/creates the CR of the Ingress Controller having Spec as follows -
+```yaml
+apiVersion: operator.openshift.io/v1
+kind: IngressController
+metadata:
+  name: default
+  namespace: openshift-ingress-operator
+spec:
+  httpHeaders:
+    headersManipulation:
+      response: 
+      - name: X-Frame-Options
+        action: Set
+        set:
+          value: DENY
+      - name: X-XSS-Protection
+        action: Delete
+```
+2. Route owner wants to set/delete HTTP response headers for only a specific route.
+   Users edits/creates the Route having Spec as follows -
+```yaml
+apiVersion: route.openshift.io/v1
+kind: Route
+metadata:
+  labels:
+    type: sharded 
+  name: hello-openshift-edge
+  namespace: hello-openshift
+spec:
+  subdomain: hello-openshift 
+  tls:
+    termination: edge
+  to:
+    kind: Service
+    name: hello-openshift
+  httpHeaders:
+    headersManipulation:
+      response:
+      - name: X-Frame-Options
+        action: Set
+        set:
+          value: DENY
+      - name: X-XSS-Protection
+        action: Delete
+```
+
+#### Variation [optional]
+
+### API Extensions
+
+A) Set/Delete HTTP response headers in the frontend sections of haproxy.config via IngressController's Spec.
+The existing Ingress Controller API `HTTPHeaders *IngressControllerHTTPHeaders` is extended by adding a field for the header manipulations.
+
+```go
+// IngressControllerSpec is the specification of the desired behavior of the
+// IngressController.
+type IngressControllerSpec struct {
```

Comment:
Is this API up to date with the PR? I'm not going to review thoroughly as we can do that in the API PR, but there are a few comments I'd like to highlight to other reviewers

Reply:
yes it is up to date